### PR TITLE
feature:S3C-3527 list tenants-1

### PR DESCRIPTION
**Pull request by saitharunthodupunuru**
Changes:
* Moved all external service calls to a Factory class
* VaultAdmin changes with ListAccounts API
* Fix CreateTenant HttpStatus for errors. CreateTenant should allow only 400 HTTP status for errors.
* Modify helper method signature

### CODE REVIEW

File: vault-admin-client/src/main/java/com/scality/osis/vaultadmin/impl/ExternalServiceFactory.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        try {
+            com.amazonaws.Response<R> response = vaultMethod.apply(request);
+            if (null!= response.getHttpResponse() && ErrorUtils.isSuccessful(response.getHttpResponse().getStatusCode())) {
+                return response.getAwsResponse();
+            } else {
+                throw ErrorUtils.parseError(response.getHttpResponse());
+            }
+        } catch (VaultServiceException e) {
+            throw e;
+        }catch (VaultClientException e){
```
Comment (anurag4DSB) [inline vault-admin-client/src/main/java/com/scality/osis/vaultadmin/impl/ExternalServiceFactory.java:36]: space missing.
I agree with what @RachedBenMustapha said in another PR. We need a linter.

File: vault-admin-client/src/test/java/com/scality/osis/vaultadmin/impl/BaseTest.java [hunk 2]
```diff
@@ -75,6 +81,73 @@ protected void initMocks() {
...
+                List<AccountData> accounts = new ArrayList<>();
+                // Generate Accounts with ids (markerVal + i) to maxItems count
+                for(; i < maxItems; i++){
+                  AccountData data = new AccountData();
+                  data.setEmailAddress(DEFAULT_TEST_EMAIL_ADDR);
+                  data.setName(DEFAULT_TEST_ACCOUNT_NAME);
+                  data.setArn(DEFAULT_TEST_ARN_STR + DEFAULT_TEST_ACCOUNT_NAME +"/\"");
+                  data.setCreateDate(new Date());
+                  data.setId(DEFAULT_TEST_ACCOUNT_ID + (i + markerVal)); //setting ID with index
+                  data.setCanonicalId(DEFAULT_TEST_CANONICAL_ID);
```
Comment (anurag4DSB) [inline vault-admin-client/src/test/java/com/scality/osis/vaultadmin/impl/BaseTest.java:128]: You might get another field if quota is set.
It only shows up in listAccounts if quota is set other than zero.
Just like `customAttributes`

Comment (anurag4DSB) [reply vault-admin-client/src/test/java/com/scality/osis/vaultadmin/impl/BaseTest.java:128]: example:
```js
❯ ./node_modules/vaultclient/bin/vaultclient --config ../admin.conf create-account --name=an2usdr2ag2 --email ans2durag2@test.com  --quota 100 --host 127.0.0.1 --port 8600
{
    "account": {
        "arn": "arn:aws:iam::256476687020:/an2usdr2ag2/",
        "canonicalId": "24e9e24a454ee9a0112fb296fe1e4362eaf79d609961de76ccac37f65a7d9b5c",
        "id": "256476687020",
        "emailAddress": "ans2durag2@test.com",
        "name": "an2usdr2ag2",
        "createDate": "2021-03-26T11:02:52Z",
        "aliasList": [],
        "oidcpList": [],
        "quotaMax": 100,
        "customAttributes": {}
    }
}
❯ ./node_modules/vaultclient/bin/vaultclient --config ../admin.conf list-accounts   --host 127.0.0.1 --port 8600
{
    "isTruncated": false,
    "accounts": [
        {
            "arn": "arn:aws:iam::256476687020:/an2usdr2ag2/",
            "canonicalId": "24e9e24a454ee9a0112fb296fe1e4362eaf79d609961de76ccac37f65a7d9b5c",
            "id": "256476687020",
            "emailAddress": "ans2durag2@test.com",
            "name": "an2usdr2ag2",
            "createDate": "2021-03-26T11:02:52Z",
            "quota": 100
        }
    ]
}
```

**Review by anurag4DSB — COMMENTED** [review]
Need to handle quota value if required for list accounts output.

File: vault-admin-client/src/main/java/com/scality/osis/vaultadmin/impl/ExternalServiceFactory.java (hunk 1 above)
Comment (saitharunthodupunuru) [reply vault-admin-client/src/main/java/com/scality/osis/vaultadmin/impl/ExternalServiceFactory.java:36]: Included more PMD (Static code analysis / Linter) rules and fixed them in b9148c7

File: vault-admin-client/src/test/java/com/scality/osis/vaultadmin/impl/BaseTest.java (hunk 2 above)
Comment (saitharunthodupunuru) [reply vault-admin-client/src/test/java/com/scality/osis/vaultadmin/impl/BaseTest.java:138]: This is a fix on `vaultclient-java`. Created a PR for this
https://github.com/scality/vaultclient-java/pull/16

**Review by ilkescality — APPROVED** [review]

**Pull request merged**